src/ipmode.rs

```diff
@@ -0,0 +1,240 @@
+use crate::Enr;
+use std::net::SocketAddr;
+///! A set of configuration parameters to tune the discovery protocol.
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum IpMode {
+    Ip4,
+    Ip6 { enable_mapped_addresses: bool },
+}
```

Comment:
I think we should discuss a third variant - dual stack. 
As I understand, these are mutually exclusive and we might partition the network. A dual stack node might be able to bridge the two, but it will be a bit more work to add this in.

Reply:
Age, dual stack is ipv6 with mapped addresses. There are 3 variants